image/default_image.go

```diff
@@ -122,6 +123,76 @@ func (d DefaultImageService) Login(RegistryURL, RegistryUsername, RegistryPasswd
 	return nil
 }
 
+func (d DefaultImageService) Delete(imageName string) error {
+	var (
+		images []*v1.Image
+		image  *v1.Image
+	)
+	named, err := reference.ParseToNamed(imageName)
+	if err != nil {
+		return err
+	}
+
+	image, err = imageutils.GetImage(named.Raw())
+	if err != nil {
+		return err
+	}
+
+	imageMetadataMap, err := imageutils.GetImageMetadataMap()
+	if err != nil {
+		return err
+	}
+	for _, imageMetadata := range imageMetadataMap {
+		if imageMetadata.ID == "" {
+			continue
+		}
+		tmpImage, err := imageutils.GetImageByID(imageMetadata.ID)
+		if err != nil {
+			continue
+		}
+		images = append(images, tmpImage)
+	}
+	layer2ImageNames := layer2ImageMap(images)
+	layerStore, err := store.NewDefaultLayerStore()
+	if err != nil {
+		return err
+	}
+	for _, layer := range image.Spec.Layers {
+		layerID := store.LayerID(digest.NewDigestFromEncoded(digest.SHA256, layer.Hash))
+		if isLayerDeletable(layer2ImageNames, layerID) {
+			err = layerStore.Delete(layerID)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = d.deleteImageLocal(image.Name, image.Spec.ID)
```

Comment:
deleting image before deleting could be better? in case that some of layers deletion failed